vic/drivers/image/src/vic_force.c

```diff
@@ -385,3 +385,83 @@ vic_force(void)
     // cleanup
     free(dvar);
 }
+
+/******************************************************************************
+ * @brief    Determine timestep and start year, month, day, and seconds of forcing files
+ *****************************************************************************/
+void
+get_forcing_file_info(param_set_struct *param_set,
+                      size_t            file_num)
+{
+    extern global_param_struct global_param;
+    extern filenames_struct    filenames;
+
+    double                     nc_times[2];
+    double                     nc_time_origin;
+    size_t                     start = 0;
+    size_t                     count = 2;
+    char                      *nc_unit_chars = NULL;
+    char                      *calendar_char = NULL;
+    unsigned short int         time_units;
+    unsigned short int         calendar;
+    dmy_struct                 nc_origin_dmy;
+    dmy_struct                 nc_start_dmy;
+
+    // read time info from netcdf file
+    get_nc_field_double(filenames.forcing[0], "time", &start, &count, nc_times);
+    get_nc_var_attr(filenames.forcing[0], "time", "units", &nc_unit_chars);
+    get_nc_var_attr(filenames.forcing[0], "time", "calendar", &calendar_char);
+
+    // parse the calendar string and check to make sure it matches the global clock
+    calendar = calendar_from_chars(calendar_char);
+
+    // parse the time units
+    parse_nc_time_units(nc_unit_chars, &time_units, &nc_origin_dmy);
+
+    // Get date/time of the first entry in the forcing file.
+    nc_time_origin = date2num(0., &nc_origin_dmy, 0., calendar, time_units);
+    num2date(nc_time_origin, nc_times[0], 0., calendar, time_units,
+             &nc_start_dmy);
+
+    // Assign file start date/time
+    global_param.forceyear[file_num] = nc_start_dmy.year;
+    global_param.forcemonth[file_num] = nc_start_dmy.month;
+    global_param.forceday[file_num] = nc_start_dmy.day;
+    global_param.forcesec[file_num] = nc_start_dmy.dayseconds;
+
+    // calculate timestep in forcing file
+    if (time_units == TIME_UNITS_DAYS) {
+        param_set->force_steps_per_day[file_num] =
+            (size_t) nearbyint(1. / (nc_times[1] - nc_times[0]));
```

Comment:
Well, we have to do some rounding here because `nc_times` is type `double` and we want to know how many steps per day are in the forcing file.  We do raise an error if they do not match (later in `get_global_param.c`).